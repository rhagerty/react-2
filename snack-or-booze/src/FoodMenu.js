import React from "react";
import { Link } from "react-router-dom";
import "./FoodMenu.css";
import {
  Card,
  CardBody,
  CardTitle,
  CardText,
  ListGroup,
  ListGroupItem
} from "reactstrap";

function FoodMenu({ snacks, title }) {
  
  return (
    <section className="col-md-4">
      <Card>
        <CardBody>
          <CardTitle className="font-weight-bold text-center">
            {title} Menu
          </CardTitle>
          <CardText>
            Below is a list of our current menu options.
          </CardText>
          <ListGroup>
            {snacks.map(snack => (
              <Link to={`/${title}/${snack.id}`} key={snack.id}>
                <ListGroupItem>{snack.name}</ListGroupItem>
              </Link>
            ))}
          </ListGroup>
          <CardText>
            Submit your own recipe <Link to={`/create/${title}`}><button>here!</button></Link>
          </CardText>
        </CardBody>
      </Card>
    </section>
  );
}

export default FoodMenu;
