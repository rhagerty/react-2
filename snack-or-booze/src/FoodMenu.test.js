import React from 'react';
import FoodMenu from './FoodMenu';
import {render} from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';


let snacks = [
    {
      "id": "nachos",
      "name": "Nachos",
      "description": "An American classic!",
      "recipe": "Cover expensive, organic tortilla chips with Cheez Whiz.",
      "serve": "Serve in a hand-thrown ceramic bowl, garnished with canned black olives"
    }]
// basic tests
it('renders without crashing', () => {
  render(
    <MemoryRouter>
      <FoodMenu snacks={snacks} title={"snacks"} />
    </MemoryRouter>
  );
});

it('matches snapshot', function() {
  const container = render(
    <MemoryRouter>
      <FoodMenu snacks={snacks} title={"snacks"} />
    </MemoryRouter>
  );
  expect(container.asFragment()).toMatchSnapshot();
});
// end basic tests

// full render
it('mounts without crashing', function() {
  const { getByText } = render(
    <MemoryRouter>
      <FoodMenu snacks={snacks} title={"snacks"} />
    </MemoryRouter>
  );

  const blogLink = getByText(/create/i);
  expect(blogLink).toBeInTheDocument();
  expect("nachos").toBeInTheDocument()
});
