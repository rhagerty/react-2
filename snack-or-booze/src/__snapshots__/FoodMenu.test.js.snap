// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <section
    class="col-md-4"
  >
    <div
      class="card"
    >
      <div
        class="card-body"
      >
        <h5
          class="font-weight-bold text-center card-title"
        >
          snacks Menu
        </h5>
        <p
          class="card-text"
        >
          Below is a list of our current menu options.
        </p>
        <ul
          class="list-group"
        >
          <a
            href="/snacks/nachos"
          >
            <li
              class="list-group-item"
            >
              Nachos
            </li>
          </a>
        </ul>
        <p
          class="card-text"
        >
          Submit your own recipe 
          <a
            href="/create/snacks"
          >
            <button>
              here!
            </button>
          </a>
        </p>
      </div>
    </div>
  </section>
</DocumentFragment>
`;
